### PR TITLE
Fix bug for fixing comma placements with inline comment

### LINKED ISSUE #799: L019 (trailing vs leading commas) fix doesn't flow around inline comments
## Expected Behaviour

Inline comments should be taken into consideration when fixing leading to trailing

## Observed Behaviour

The fix incorrectly places a comma _after_ the inline comment, instead of after the field and _before_ the comment

## Steps to Reproduce

`test.sql`
```sql
select
    field_1,
    field_2 -- some comment,
    field_3
from my_table
```

```
$ sqlfluff fix test.sql
==== finding fixable violations ====
== [test_sql.sql] FAIL
L:   3 | P:   5 | L019 | Found leading comma. Expected only trailing.
L:   4 | P:   5 | L019 | Found leading comma. Expected only trailing.
==== fixing violations ====
2 fixable linting violations found
Are you sure you wish to attempt to fix these? [Y/n] ...
Attempting fixes...
Persisting Changes...
== [test_sql.sql] PASS
Done. Please check your files to confirm.
```

fixed `test.sql`
```sql
select
    field_1,
    field_2 -- some comment,
    field_3
from my_table
```

## Version
Include the output of `sqlfluff --version` along with your Python version
sqlfluff, version 0.4.0
Python 3.8.6
## Configuration

None

This one is on me 🤦🏻 I didn't consider this test case. I'll submit a PR with a failing test case soon

### PR DESCRIPTION
**Pull request by pwildenhain**
Fixes #799

### CODE REVIEW

File: test/fixtures/rules/std_rule_cases/L019.yml [hunk 1]
```diff
@@ -12,6 +12,24 @@ leading_comma_violations:
...
+        a
+        , b -- inline comment
+        , c
+        /* non inline comment */
+        , d
+    FROM e
+  fix_str: |
+    SELECT
+        a,
+        b ,-- inline comment
```
Comment (barrywhart) [inline test/fixtures/rules/std_rule_cases/L019.yml:27]: The formatting looks odd here -- space before the comma, no space between the comma and the comment. Will the final output look like this, or do other lint rules/fixes typically handle this?

Comment (pwildenhain) [reply test/fixtures/rules/std_rule_cases/L019.yml:27]: Yeah, I said more about this in 3bdf0583fe78cedec1b5623ab562a43542231c36 but basically I had a hard time figuring out the formatting. The final output looks normal though, as other rules get applied. If you have some suggestions though I'm all ears 👂

Comment (barrywhart) [reply test/fixtures/rules/std_rule_cases/L019.yml:27]: No suggestions, really, although I'm sure it's doable with more tinkering. We were discussing a similar "rule interdependencies" question on my open PR #905. I think I'm okay having other rules help out with whitespace formatting, but I'm not sure Niall and Alan are convinced. 😜

Comment (pwildenhain) [reply test/fixtures/rules/std_rule_cases/L019.yml:27]: Thanks for the heads up, let's see what they have to say 🤔

https://github.com/sqlfluff/sqlfluff/pull/905#discussion_r609607938

Comment (alanmcruickshank) [reply test/fixtures/rules/std_rule_cases/L019.yml:27]: Until we resolve the other conversation I'm fine with this. This isn't the only example of where there's some odd formatting around, so I don't feel the need to hold this back until that's resolved. 👍

**Review by alanmcruickshank — APPROVED** [review]
Aside from resolving he formatting issue (which we should come back to later), this looks good, 👍

**Pull request merged**